documentation/src/main/asciidoc/topics/json/server_aggregate_realm.json

```diff
@@ -0,0 +1,45 @@
+{
+  "server": {
+    "security": {
+      "security-realms": [
+        {
+          "name": "aggregate-realm",
+          "default-realm": "aggregate",
+          "server-identities": {
+            "ssl": {
+              "keystore": {
+                "path": "server.p12",
+                "relative-to": "infinispan.server.config.path",
+                "keystore-password": "secret",
+                "alias": "server"
+              },
+              "truststore": {
+                "path": "trust.p12",
+                "relative-to": "infinispan.server.config.path",
+                "password": "secret"
+              }
+            }
+          },
+          "properties-realm": {
+            "groups-attribute": "Roles",
+            "user-properties": {
+              "digest-realm-name": "distributed-realm",
+              "path": "users.properties"
+            },
+            "group-properties": {
+              "path": "groups.properties"
+            }
+          },
+          "truststore-realm": {},
+          "aggregate-realm": {
+            "authentication-realm": "trust",
+            "authorization-realms": ["properties"],
```

Comment:
So `trust` and `properties` are used here because they are the default values for the NAME property in each of the realms right? I don't think we handle the following scenario:

```json
	"properties-realm": {
		"name": "would-anyone-actually-do-this",
		"groups-attribute": "Roles",
		"user-properties": {
			"digest-realm-name": "distributed-realm",
			"path": "users.properties"
		},
		"group-properties": {
			"path": "groups.properties"
		}
	},
	"truststore-realm": {
		"name": "would-anyone-actually-do-this"
	}
```

We should throw an exception if two `RealmProvider` instances exist with the same name.

Reply:
It's also not clear from the documentation that the value passed to `authentication-realm` and `authorization-realms` should be the names of the realm. In the example above, a user could just assume that was shorthand for the "truststore-realm" element name (I had to check in the code).

Reply:
We certainly handle duplicate errors: https://github.com/infinispan/infinispan/blob/main/server/runtime/src/main/java/org/infinispan/server/configuration/security/RealmConfigurationBuilder.java#L97

Reply:
Great! I was looking in the wrong places :slightly_smiling_face: